examples/gallery/lines/line_segment_ends.py

```diff
@@ -0,0 +1,75 @@
+"""
+Line segment caps and joints
+============================
+PyGMT offers different appearance of line segment caps and joints. The desired
+appearance can be set via the GMT default parameters :gmt-term:`PS_LINE_CAP`
+(``"butt"``, ``"round"``, or ``"square"`` [Default]) as well as :gmt-term:`PS_LINE_JOIN`
+(``"bevel"``, ``"round"``, and ``"miter"`` [Default]) and :gmt-term:`PS_MITER_LIMIT`
+(limit on the angle at the mitered joint below which a bevel is applied).
+"""
+
+# %%
+import numpy as np
+import pygmt
+
+# Set up dictionary for colors
+dict_col = {
+    "round": "green4",
+    "square": "steelblue",
+    "butt": "orange",
+    "miter": "steelblue",
+    "bevel": "orange",
+}
+
+# Create new Figure instance
+fig = pygmt.Figure()
+
+# -----------------------------------------------------------------------------
+# Top: PS_LINE_CAP
+
+# Create sample data
+x = np.array([30, 170])
+y = np.array([70, 70])
+
+fig.basemap(region=[0, 250, 0, 100], projection="x1p", frame="rltb")
+
+for line_cap in ["butt", "round", "square"]:
+    # Change GMT default locally using a context manager
+    # The change applies only to the code under the with statement
+    with pygmt.config(PS_LINE_CAP=line_cap):
+        # Plot dashed line
+        color = dict_col[line_cap]
+        fig.plot(x=x, y=y, pen=f"10p,{color},20_20:0")
+
+    fig.plot(x=x, y=y, pen="1p")
+    fig.plot(x=x, y=y, style="c0.1c", fill="white", pen="0.5p")
+    fig.text(text=line_cap, x=x[-1] + 20, y=y[-1], justify="LM")
+
+    y = y - 20
+
+fig.shift_origin(yshift="-h")
+
+# -----------------------------------------------------------------------------
+# Bottom: PS_LINE_JOIN and PS_MITER_LIMIT
+
+# Create sample data
+x = np.array([[5, 95, 65], [80, 170, 140], [155, 235, 205]])
+y = np.array([[10, 70, 10], [10, 70, 10], [10, 70, 10]])
```

Comment:
This sample data is a little too complicated. Maybe just define the data as
```
x = np.array([5, 95, 65])
y = np.array([10, 70, 10])
```
for other two lines, just use `x += 75`.

Reply:
Yes, you are completely right! I totally overlooked this after changing to lines which are only shifted in x-direction in the second part! Done in https://github.com/GenericMappingTools/pygmt/pull/3015/commits/bdaf946b4b22323feecfe17c23f7975b5be9f41a.